python/lsst/pipe/tasks/match_tract_catalog.py

```diff
@@ -0,0 +1,222 @@
+        name="{name_input_cat_target}",
+        storageClass="DataFrame",
+        dimensions=("tract", "skymap"),
+        deferLoad=True,
+    )
+    skymap = cT.Input(
+        doc="Input definition of geometry/bbox and projection/wcs for coadded exposures",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+    cat_output_ref = cT.Output(
+        doc="Reference matched catalog with indices of target matches",
+        name="match_ref_{name_input_cat_ref}_{name_input_cat_target}",
+        storageClass="DataFrame",
+        dimensions=("tract", "skymap"),
+    )
+    cat_output_target = cT.Output(
+        doc="Target matched catalog with indices of reference matches",
+        name="match_target_{name_input_cat_ref}_{name_input_cat_target}",
+        storageClass="DataFrame",
+        dimensions=("tract", "skymap"),
+    )
+
+
+class MatchTractCatalogSubConfig(pexConfig.Config):
+    """Config class for the MatchTractCatalogSubTask to define methods returning
+    values that depend on multiple config settings.
+    """
+    @property
+    @abstractmethod
+    def columns_in_ref(self) -> Set[str]:
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def columns_in_target(self) -> Set[str]:
+        raise NotImplementedError()
+
+
+class MatchTractCatalogSubTask(pipeBase.Task, ABC):
+    """An abstract interface for subtasks of MatchTractCatalogTask to match
+    two tract object catalogs.
+
+    Parameters
+    ----------
+    **kwargs
+        Additional arguments to be passed to the `lsst.pipe.base.Task`
+        constructor.
+    """
+    ConfigClass = MatchTractCatalogSubConfig
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    @abstractmethod
+    def run(
+        self,
+        catalog_ref: pd.DataFrame,
+        catalog_target: pd.DataFrame,
+        wcs: afwGeom.SkyWcs = None,
+    ) -> pipeBase.Struct:
+        """Match sources in a reference tract catalog with a target catalog.
+
+        Parameters
+        ----------
+        catalog_ref : `pandas.DataFrame`
+            A reference catalog to match objects/sources from.
+        catalog_target : `pandas.DataFrame`
+            A target catalog to match reference objects/sources to.
+        wcs : `lsst.afw.image.SkyWcs`
+            A coordinate system to convert catalog positions to sky coordinates.
+
+        Returns
+        -------
+        retStruct : `lsst.pipe.base.Struct`
+            A struct with output_ref and output_target attribute containing the
+            output matched catalogs.
+        """
+        raise NotImplementedError()
+
+
+class MatchTractCatalogConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=MatchTractCatalogConnections,
+):
+    """Configure a MatchTractCatalogTask, including a configurable matching subtask.
+    """
+    match_tract_catalog = pexConfig.ConfigurableField(
```

Comment:
Ah okay, here's the configurable. Disregard my comment about making this a subtask.